File: problem_two/2.js
function generateFibos (term, fibos, i) {
  fibos = fibos || [1,2];
  i = i || 0;
  var new_fibo = fibos[i] + fibos[i+1];
  if (new_fibo >= term) {
    return fibos;
  }
  fibos.push(new_fibo);
  return generateFibos(term, fibos, i+1);
}

function getEvenFibos (term) {
  var fibos = generateFibos(term);
  var evenFibos = [];
  for (var i=0; i<fibos.length; i++){
    if (isEven(fibos[i])) {
      evenFibos.push(fibos[i]);
    }
  }
  return evenFibos;
}

function isEven (number) {
  return (number % 2 ===0);
}

(function doMagic (term) {
  var evenFibos = getEvenFibos(term);
  var result = evenFibos.reduce(function(i,j){
    return i + j;
  });
  return console.log(result);
})(4000000);

File: problem_two/2.rb
def generate_fibo(term, fibos=[1,2], i=0)
  new_fibo = fibos[i] + fibos[i+1]
  return fibos if new_fibo >= term
  fibos << new_fibo
  generate_fibo(term, fibos, i+1)
end

def do_magic(term)
  generate_fibo(term).select do |fibo|
    fibo.even?
  end.inject(:+)
end

p do_magic(4000000)